default.nix

```diff
@@ -1,20 +1,13 @@
 { system ? builtins.currentSystem
 , profiling ? false
 , iosSdkVersion ? "10.2"
+, __useLegacyCompilers ? false
 }:
 let
-  getReflexPlatform = sys: import ./dep/reflex-platform { inherit iosSdkVersion; system = sys; enableLibraryProfiling = profiling; };
-  reflex-platform = getReflexPlatform system;
-  inherit (reflex-platform) hackGet nixpkgs;
-  pkgs = nixpkgs;
-in with pkgs.haskell.lib; with pkgs.lib;
-let
-  # TODO: Remove this after updating nixpkgs: https://github.com/NixOS/nixpkgs/issues/37750
-  justStaticExecutables' = drv: let
-      drv' = justStaticExecutables drv;
-    in if pkgs.stdenv.isDarwin
-      then removeConfigureFlag drv' "--ghc-option=-optl=-dead_strip"
-      else drv';
+  cleanSource = builtins.filterSource (name: _: let baseName = builtins.baseNameOf name; in !(
```

Comment:
Btw, it's better to use `lib.cleanSourceWith` than `builtins.filterSource`. Does the same thing, but it composes with itself, whereas builtins.filterSource cannot.

Reply:
This can actually be important if you want to filter a source provided by a user, because the user may want to pass you a source they've already filtered.

Reply:
Good point. I think I'll do that in a follow-up PR if you don't mind? (this is just moving it around.)

For one, I need a better store for the inner overlay (haskell) accessing the outer `self` (`pkgs`) than this nasty `callPackage` hack!